src/store/useMiningStore.ts

```diff
@@ -35,17 +44,106 @@ const initialState: State = {
     miningInitiated: false,
     isChangingMode: false,
     miningControlsEnabled: true,
+    cpu: {
+        hardware: undefined,
+        mining: {
+            is_mining: false,
+            hash_rate: 0,
+            estimated_earnings: 0,
+            connection: { is_connected: false },
+        },
+    },
+    gpu: {
+        hardware: undefined,
+        mining: {
+            is_mining: false,
+            hash_rate: 0,
+            estimated_earnings: 0,
+        },
+    },
+    base_node: {
+        block_height: 0,
+        block_time: 0,
+        is_synced: false,
+    },
 };
 
-export const useMiningStore = create<MiningStoreState>()((set) => ({
+export const useMiningStore = create<MiningStoreState>()((set, getState) => ({
     ...initialState,
-    setDisplayBlockTime: (displayBlockTime) => set({ displayBlockTime }),
-    setDisplayBlockHeight: (displayBlockHeight) => set({ displayBlockHeight }),
-    setEarnings: (earnings) => set({ earnings }),
+    fetchMiningMetrics: async () => {
+        console.info('Fetching mining metrics...');
```

Comment:
do we need to log this every second? 😅 

also, related to this in `useMiningMetricsUpdater` - i noticed my cpu mining almost never works, so if i disabled gpu mining and hit 'start' the `fetchMiningMetrics` just goes on foreeeverr and my button is stuck on loading. 
maybe it'd help to check here which mining mode is enabled and if the metrics response ever includes 'is_mining' then stop the interval and reset mining state after like 100 tries? just an idea (i'm not sure what the best approach for that might be maybe @stringhandler has some ideas:D?)

Reply:
Was this happening before for you?

Reply:
yeah the cpu_mining has always been spotty for me locally, it either takes ages to start up or never does. so that wasn't introduced here, just think it may be a good time to add a fallback/fail on too many retries?